src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,121 @@
+package duke;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+
+/**
+ * Class for parsing user inputs to be executed
+ */
+public class Parser {
+    /**
+     * Parses user input and returns true if program is to be exited
+     * @param input user input
+     * @param lst Tasklist of Duke program
+     * @param ui user interface
+     * @param storage storage for data
+     * @return a boolean to decide if program is to be exited
+     * @throws DukeException
+     */
+    public static String parse(String input, TaskList lst, Ui ui, Storage storage) throws DukeException {
+
+        String[] words = input.split(" ", 2);
+        String first = words[0];
+
+        try {
+            if (first.equals("bye")) {
```

Comment:
Perhaps the SLAP principle could be used here (using switch-case and abstracting all the logic into separate functions)?